components/CampusToggleButton.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import Colors from '../constants/Colors';
+import { TouchableOpacity, View, StyleSheet, Text } from "react-native";
+
+class CampusToggleButton extends Component {
+
+    state = {
+        currentCampusView: 'SGW',
+    };
+
+    buttonStyling(campusSelected) {
+        const { currentCampusView } = this.state;
+        return {
+            backgroundColor: (campusSelected == currentCampusView) ? Colors.primaryColor : Colors.grey,
+            opacity: (campusSelected == currentCampusView) ? 1 : 0.7,
+            padding: 7,
+            width: 100,
+        }
+    }
+
+    buttonTextStyling(campusSelected) {
+        const { currentCampusView } = this.state;
+        return {
+            color: 'white',
+            fontWeight: 'bold',
+            fontSize: 18,
+            textAlign: 'center',
+        }
+    }
+
+    toggleCampusView(campusSelected) {
+        const { setMapLocation } = this.props;
```

Comment:
I get an error of this form/type from my ide:

`TS2339: Property 'setMapLocation' does not exist on type 'Readonly<{}> & Readonly<{ children?: ReactNode; }>'.`

It still compiles and runs since its only a ts error but leaves a red underline. I dont mind it. just pointing it out

Reply:
Yeah I get that on my IDE as well. Since it compiled I figured that maybe its something in my configurations. Its something to investigate into